gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/BaseKey.java

```diff
@@ -172,20 +172,21 @@ public boolean equals(Object obj) {
   }
 
   @Override
-  protected DatastoreV1.Key toPb() {
-    DatastoreV1.Key.Builder keyPb = DatastoreV1.Key.newBuilder();
-    DatastoreV1.PartitionId.Builder partitionIdPb = DatastoreV1.PartitionId.newBuilder();
-    if (projectId != null) {
-      partitionIdPb.setDatasetId(projectId);
+  protected com.google.datastore.v1beta3.Key toPb() {
+    com.google.datastore.v1beta3.Key.Builder keyPb = com.google.datastore.v1beta3.Key.newBuilder();
+    com.google.datastore.v1beta3.PartitionId.Builder partitionIdPb = 
+        com.google.datastore.v1beta3.PartitionId.newBuilder();
+    if (!Strings.isNullOrEmpty(projectId)) {
```

Comment:
I'm going to actually change this back to `(projectId != null)`, there isn't a need for it to change.  Same for `if` statement just below this one.